### ISSUE TITLE
Error when setting preset mode in UI

### ISSUE THREAD

**Issue by msp1974** [asantaga/wiserHomeAssistantPlatform#300]
When setting preset mode via UI more info panel, an error occurs that preset_mode is None, 

```
Failed to call service climate/set_preset_mode. String value is None for dictionary value @ data['preset_mode']
```

however it does set the preset mode fine.

Upon investigation, this is caused by the value we set preset mode to after selection not being in the list of preset modes as we use this to show boost time remaining and hub modes that cannot be set by selection in a room.

Need to review how to resolve this and keep same functionality or this could be a signidicant breaking change.

**Comment by msp1974** [comment]
Further investigation shows that UI calls the set_preset_mode service again if mode set by integration and more-info card is being shown.  Have raised with HA frontend as issue #13801.

**Comment by msp1974** [comment]
Have submitted a PR to HA frontend to resolve this issue.  PR #13825

**Comment by msp1974** [comment]
This was fixed in HA 2022.10.0.  Closing issue.
